arch/tools/cli/core.py

```diff
@@ -1,40 +1,73 @@
 import subprocess
 import os
 import time
-import pkg_resources
-import select
 import sys
 import glob
-from cli.utils import run_docker_compose_ps, print_service_status, check_services_state
+import docker
 from cli.utils import getLogger
 from cli.consts import (
+    ARCHGW_DOCKER_IMAGE,
+    ARCHGW_DOCKER_NAME,
     KATANEMO_LOCAL_MODEL_LIST,
     MODEL_SERVER_LOG_FILE,
     ACCESS_LOG_FILES,
 )
 from huggingface_hub import snapshot_download
+from dotenv import dotenv_values
+
 
 log = getLogger(__name__)
 
+dot_env = dotenv_values(".env")
+
+
+def start_archgw_docker(client, arch_config_file):
+    return client.containers.run(
+        name=ARCHGW_DOCKER_NAME,
+        image=ARCHGW_DOCKER_IMAGE,
+        detach=True,  # Run in detached mode
+        ports={
+            "10000/tcp": 10000,
+            "10001/tcp": 10001,
+            "11000/tcp": 11000,
+            "12000/tcp": 12000,
+            "19901/tcp": 19901,
+        },
+        volumes={
+            f"{arch_config_file}": {
+                "bind": "/app/arch_config.yaml",
```

Comment:
does this require someone to checkout main to run archgw?

Reply:
no, its part of archgw cli